public/js/index.js

```diff
@@ -128,6 +128,18 @@ $(function () {
         console.error("Error saving config:", strError);
       }
     });
+
+  })
+
+  $('#updateSoftware').on('click', function () {
+    if (objSocket.connected) {
+      objSocket.emit('control_hat', '{“set_charger”: “ON”}')
```

Comment:
Not sure how they got in there, but check the characters around set_charger and ON — they're not quotation marks. JSON.parse() does not like them at all.

Reply:
LOL oops.